holidays/countries/united_states.py

```diff
@@ -88,19 +88,28 @@ class UnitedStates(HolidayBase):
         "WY",
     ]
 
+    def _add_with_observed(
+        self, dt: date, name: str, before: bool = True, after: bool = True
+    ) -> None:
+        self[dt] = name
+        if not self.observed:
+            return
+        if dt.weekday() == SAT and before:
+            self[dt + rd(days=-1)] = f"{name} (Observed)"
+        elif dt.weekday() == SUN and after:
+            self[dt + rd(days=+1)] = f"{name} (Observed)"
+
     def _populate(self, year):
         super()._populate(year)
 
         # New Year's Day
-        if year > 1870:
+        if year >= 1871:
             name = "New Year's Day"
-            self[date(year, JAN, 1)] = name
-            if self.observed and date(year, JAN, 1).weekday() == SUN:
-                self[date(year, JAN, 1) + rd(days=+1)] = name + " (Observed)"
+            self._add_with_observed(date(year, JAN, 1), name, False, True)
```

Comment:
I believe it'd look more readable this way:
```suggestion
            self._add_with_observed(date(year, JAN, 1), name, before=False)
```

Reply:
Done. Thank you!